cloudinit/config/schema.py

```diff
@@ -714,18 +724,22 @@ def validate_cloudconfig_schema(
         against the provided schema.
     @raises: RuntimeError when provided config sourced from YAML is not a dict.
     @raises: ValueError on invalid schema_type not in CLOUD_CONFIG or
-        NETWORK_CONFIG
+        NETWORK_CONFIG_V1 or NETWORK_CONFIG_V2
     """
     if schema_type == SchemaType.NETWORK_CONFIG:
-        if network_schema_version(config) == 2:
-            if netplan_validate_network_schema(
-                network_config=config, strict=strict, log_details=log_details
-            ):
-                # Schema was validated by netplan
-                return True
-            # network-config schema version 2 but no netplan.
-            # TODO(add JSON schema definition for network version 2)
-            return False
+        network_version = network_schema_version(config)
+        if network_version == 2:
+            schema_type = SchemaType.NETWORK_CONFIG_V2
+        elif network_version == 1:
+            schema_type = SchemaType.NETWORK_CONFIG_V1
+        schema = get_schema(schema_type)
```

Comment:
Yes it does since in the places it's called within this code, schema is incorrectly set to V2 so it's important to reset it after the version has been dynamically determined from the actual config.